### ISSUE TITLE
[ros2topic] ros2 topic pub does not seem to parse arrays of messages

### ISSUE THREAD

**Issue by mxgrey** [ros2/ros2cli#191]
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 18.04
- Installation type:
  - Source
- Version or commit hash:
  - 7a65e1ecf02c69bf3d9624d49b17cde57f7ed16a
- DDS implementation:
  - Fast-RTPS
- Client library (if applicable):
  - N/A

#### Steps to reproduce issue

Run the following from the command line

```
$ ros2 topic pub some_pose_array geometry_msgs/PoseArray "{header: {stamp: {sec: 1, nanosec: 2}, frame_id: some_frame}, poses: [{position: {x: 0, y: 1, z: 2}, orientation: {x: 3, y: 4, z: 5, w: 6}}]}"
```

#### Expected behavior

The message gets published

#### Actual behavior

The console prints:

```
Failed to populate field 'poses': The 'poses' field must be a set or sequence and each value of type 'Pose'
```

#### Additional information

The specific example above isn't important, but it seems that `ros2 topic pub` cannot currently parse arrays of message types.

**Comment by jubeira** [comment]
@cottsay would you please assign this one to me? Thanks in advance!

**Comment by mikaelarguedas** [comment]
This looks like a duplicate of https://github.com/ros2/ros2cli/issues/59

I submitted a fix at https://github.com/ros2/ros2cli/pull/197 if you guys want to give it a try

**Comment by mxgrey** [comment]
Sorry for duplicating the issue! You're right; it's the same problem. I'll close this to avoid duplicated tickets, and I'll test the fix at the first chance that I get.

**Comment by jubeira** [comment]
Just for the record, I tested the fix with #197 and https://github.com/ros2/rosidl_python/pull/30 and it worked fine :+1:
